server/hl7/hl7.controller.js

```diff
@@ -117,6 +117,24 @@ function getUserFiles(req, res, next) {
   return next(err);
 }
 
+/**
+ * Get get a file given it's id
+ * @returns {file}
+ */
+function getFile(req, res, next) {
+  const currentUser = req.user;
+  const { fileId } = req.params
+  const file = currentUser.files.find((f) => {
+    return f._id === fileId
```

Comment:
```suggestion
    return f._id.toString() === fileId
```
Changing to the strict `===` revealed something. `_id` is an ObjectID, not a string, so `===` returns false while `==` pretends that the ObjectID is a string and returns true. Calling the ObjectID's `toString` lets us do a `===` string comparison between the id and the path parameter.

Reply:
So it seems like in every case (whether the `fileId` passed in is valid or not) we hit this return statement and get back a `204 No Content`. Not sure if this is the intended behavior